tools/ds-identify

```diff
@@ -179,10 +179,30 @@ debug() {
     echo "$@" 1>&3
 }
 
+map_sysfs_to_dmi() {
+    case "$@" in
+        sys_vendor) _RET="system-manufacturer";;
+        product_name) _RET="system-product-name";;
+        product_uuid) _RET="system-uuid";;
+        product_serial) _RET="system-serial-number";;
+        chassis-asset-tag) _RET="chassis_asset_tag";;
+        *) return 1;;
+    esac
+}
+
 get_dmi_field() {
     local path="${PATH_SYS_CLASS_DMI_ID}/$1"
     if [ ! -f "$path" ] || [ ! -r "$path" ]; then
-        _RET="$UNAVAILABLE"
+        map_sysfs_to_dmi "$1" || {
+            _RET="$ERROR: unknown field $1"
+            return
+        }
+        local dmi_field="$_RET" dmi_value=""
+        dmi_value=$(dmidecode --quiet --string "$dmi_field") || {
```

Comment:
You can also use kenv, for that and the parsing may be easier:

```
[goneri@freebsd-10 ~]$ kenv|grep smbios
smbios.bios.reldate="04/01/2014"
smbios.bios.vendor="SeaBIOS"
smbios.bios.version="?-20190727_073836-buildvm-ppc64le-16.ppc.fedoraproject.org-3.fc31"
smbios.chassis.maker="QEMU"
smbios.chassis.version="pc-i440fx-4.1"
smbios.memory.enabled="786432"
smbios.socket.enabled="1"
smbios.socket.populated="1"
smbios.system.maker="QEMU"
smbios.system.product="Standard PC (i440FX + PIIX, 1996)"
smbios.system.uuid="344e9cd2-eafa-4fb4-bdd4-ec4ede518d4f"
smbios.system.version="pc-i440fx-4.1"
smbios.version="2.8
```

Reply:
yah, but as it is, this will work on any system that's not Linux since `dmidecode` is platform independent, `kenv` probably is not

Reply:
dmidecode is OS independent, but not platform independent.
it only runs on x86 derivates and on aarch64 (but not on FreeBSD)